library/alloc/src/collections/btree/node.rs

```diff
@@ -483,18 +494,49 @@ impl<'a, K, V, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
         unsafe { self.reborrow_mut().into_val_mut_at(idx) }
     }
 
-    fn keys_mut(&mut self) -> &mut [K] {
+    fn keys_mut(&mut self) -> &mut [K]
+    where
+        K: 'a,
+        V: 'a,
```

Comment:
The need for this bound is created by inlining into_key_slice_mut, but don't ask me why.